pkg/blockstorage/vmware/vmware.go

```diff
@@ -242,10 +242,25 @@ func (p *FcdProvider) SnapshotDelete(ctx context.Context, snapshot *blockstorage
 		log.Debug().Print("Started SnapshotDelete task", field.M{"VolumeID": volID, "SnapshotID": snapshotID})
 		_, lerr = task.Wait(ctx, defaultWaitTime)
 		if lerr != nil {
-			if strings.Contains(lerr.Error(), invalidStateError) {
-				log.Debug().Print("Retrying SnapshotDelete task", field.M{"VolumeID": volID, "SnapshotID": snapshotID})
-				// retry operation
-				return false, nil
+			// The following error handling was pulled from https://github.com/vmware-tanzu/astrolabe/blob/91eeed4dcf77edd1387a25e984174f159d66fedb/pkg/ivd/ivd_protected_entity.go#L433
+			if soap.IsVimFault(lerr) {
+				switch soap.ToVimFault(err).(type) {
+				case *types.InvalidArgument:
+					log.Debug().WithError(err).Print("Disk doesn't have given snapshot due to the snapshot stamp was removed in the previous DeleteSnapshot operation which failed with InvalidState fault. And it will be resolved by the next snapshot operation on the same VM. Will NOT retry")
```

Comment:
Grammar!
- "... due to the snapshot stamp being removed..." or "... because the snapshot stamp was removed ..."
- "with _an_ InvalidState fault"
- Sentence starts with "And"

Debug errors are useless as K10 eats them up. Please change to Error.

Reply:
So these error messages are copied verbatim. I can change them for our purposes. Will change to error.